### DISCUSSION TITLE
SSH Multiplexing from python library

### DISCUSSION THREAD

**Discussion by rinshadka** [ParallelSSH/parallel-ssh#349] [Q&A] [answered]
Hi,

I have configured SSH multiplexing easily and was able to successfully test the SSH connections and their sessions. while it's working fine from both ssh as well as parallel-ssh command line, our multiplexing configurations are not picked by the parallel-ssh python code. I have done a few debugging regarding this issue and also raised [issues.](https://github.com/ParallelSSH/parallel-ssh/issues/347) But couldn't resolve this issue from python code.

Thanks.

**Comment by pkittenis** [comment, marked as answer]
Multiplexing is not supported by this library, nor is it needed.

If you want to keep connections alive for an extended duration of time, simply keep the `client` object in scope in your code/application. All commands are run on an existing per-client-object connection if one is available. The client automatically configures keep-alive packets so that connections remain alive.

New client objects will use new connections, by design.
